src/database/test/DatabaseTests.cpp

```diff
@@ -612,105 +627,58 @@ TEST_CASE("schema upgrade test", "[db]")
 
         LedgerTxn ltx(app->getLedgerTxnRoot());
 
-        for (auto aol : accOptLiabilitiesVec)
+        for (auto ae : accountEntries)
         {
             LedgerKey key;
-            key.type(ACCOUNT);
-            key.account().accountID = aol.first.accountID;
-            auto acc = ltx.load(key);
-            REQUIRE(acc.current().data.type() == ACCOUNT);
-            REQUIRE(acc.current().data.account().accountID ==
-                    aol.first.accountID);
-            REQUIRE(acc.current().data.account().balance == aol.first.balance);
-            REQUIRE(acc.current().data.account().flags == aol.first.flags);
-            REQUIRE(acc.current().data.account().homeDomain ==
-                    aol.first.homeDomain);
-            REQUIRE(acc.current().data.account().inflationDest ==
-                    aol.first.inflationDest);
-            REQUIRE(acc.current().data.account().numSubEntries ==
-                    aol.first.numSubEntries);
-            REQUIRE(acc.current().data.account().seqNum == aol.first.seqNum);
-            REQUIRE(acc.current().data.account().signers == aol.first.signers);
-            REQUIRE(acc.current().data.account().thresholds ==
-                    aol.first.thresholds);
-            REQUIRE(acc.current().ext == LedgerEntry::_ext_t());
-            REQUIRE(acc.current().ext.v() == 0);
-            if (aol.second)
-            {
-                REQUIRE(acc.current().data.account().ext.v() == 1);
-                REQUIRE(
-                    acc.current().data.account().ext.v1().liabilities.buying ==
-                    aol.second->buying);
-                REQUIRE(
-                    acc.current().data.account().ext.v1().liabilities.selling ==
-                    aol.second->selling);
-            }
-            else
-            {
-                REQUIRE(acc.current().data.account().ext.v() == 0);
-            }
+            LedgerEntry entry;
+            entry.data.type(ACCOUNT);
+            entry.data.account() = ae;
+            key = LedgerEntryKey(entry);
+            auto ae_upgraded = ltx.load(key);
+            REQUIRE(ae_upgraded.current().data.type() == ACCOUNT);
```

Comment:
I still think you're working too hard here. It would be more compact and reliable to use
```c++
operator==(LedgerEntry const&, LedgerEntry const&)
```

Is the reason you aren't doing this that those comparisons fail because you don't have the right `lastModifiedLedgerSeq`? If so, it would probably be better to just fix the root cause. Or is it something else?

Reply:
Oh!  The comparisons that you're suggesting work; I'm just not used to the types, and it simply hadn't occurred to me that ```ae_upgraded``` (now ```aeUpgraded```) and ```entry``` were comparable.  I've made that change.